buildSrc/docker-v2.gradle

```diff
@@ -0,0 +1,134 @@
+String dockerStagingDirectory = project.buildDir.toString() + '/docker-staging'
+String baseDockerDirectory = project.projectDir.toString() + '/docker'
+
+// Names of non-image specific Docker tasks
+String dockerCleanStagingAreaDirectory = 'dockerCleanStagingAreaDirectory'
+String dockerRemoveAllImages = 'dockerRemoveAllImages'
+String dockerSetupStagingAreaDirectory = 'dockerSetupStagingAreaDirectory'
+String dockerBuildAllImages = 'dockerBuildAllImages'
+String dockerLoginDockerHub = 'dockerLoginDockerHub'
+String dockerPublishAllImages = 'dockerPublishAllImages'
+
+project.tasks.create(name: dockerCleanStagingAreaDirectory, group: 'Docker', description: 'Remove staging area used for creating Docker images') {
+    doFirst {
+        logger.lifecycle('Removing Docker staging area:: ' + dockerStagingDirectory)
+    }
+
+    file(dockerStagingDirectory).deleteDir()
+}
+clean.finalizedBy(project.tasks.findByName(dockerCleanStagingAreaDirectory))
+
+project.tasks.create(name: dockerRemoveAllImages, group: 'Docker', description: 'Remove docker images and staging area for project') {}
+
+project.tasks.create(name: dockerSetupStagingAreaDirectory, type: Copy, group: 'Docker', description: 'Populate staging area used for creating Docker images') {
+    dependsOn project.tasks.findByName(dockerCleanStagingAreaDirectory)
+    dependsOn build
+
+    doFirst {
+        logger.lifecycle('Setting up docker staging area')
+    }
+
+    into dockerStagingDirectory
+    from(baseDockerDirectory)
+
+    into('blackduck-alert') {
+        from("${project.buildDir}/distributions/blackduck-alert-boot-${project.version}.tar")
+    }
+
+    into('blackduck-alert-db') {
+        from('alert-database/src/main/resources/scripts/')
```

Comment:
Perhaps for another PR but the contents of src/main/resources/scripts can be moved to the docker directory for blackduck-alert-db docker files. They are used by the image I do not believe anything is used within the jar file.